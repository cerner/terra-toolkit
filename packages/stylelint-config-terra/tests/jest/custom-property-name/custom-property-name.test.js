/**
 * @jest-environment node
 */
const stylelint = require('stylelint');
const path = require('path');

const config = {
  plugins: [
    path.resolve(__dirname, '..', '..', '..', 'src', 'rules', 'custom-property-name', 'custom-property-name'),
  ],
  rules: {
    'terra/custom-property-name': true,
  },
};

describe('custom-property-name', () => {
  it('does not error with a matching name', () => {
    const result = stylelint.lint({
      code: '.test { color: var(--test-color, #000) }',
      config,
    });

    return result.then(data => expect(data.errored).toBeFalsy());
  });

  it('does not error with a matching name with hyphen', () => {
    const result = stylelint.lint({
      code: '.test { background-image: var(--test-background-image, #000) }',
      config,
    });

    return result.then(data => expect(data.errored).toBeFalsy());
  });

  it('does error with a mismatched name with hyphen', () => {
    const result = stylelint.lint({
      code: '.test { background-image: var(--test-icon, #000) }',
      config,
    });

    return result.then(data => {
      expect(data.errored).toBeTruthy();
      expect(data.results[0].warnings).toMatchSnapshot();
    });
  });

  it('does error with a mismatched name', () => {
    const result = stylelint.lint({
      code: '.test { color: var(--test-color-primary, #000) }',
      config,
    });

    return result.then(data => {
      expect(data.errored).toBeTruthy();
      expect(data.results[0].warnings).toMatchSnapshot();
    });
  });
});
