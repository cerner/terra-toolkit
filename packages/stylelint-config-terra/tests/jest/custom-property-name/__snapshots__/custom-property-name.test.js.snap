// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`custom-property-name does error with a mismatched name 1`] = `
Array [
  Object {
    "column": 9,
    "line": 1,
    "rule": "terra/custom-property-name",
    "severity": "error",
    "text": "Expected --test-color-primary to be suffixed with -color (terra/custom-property-name)",
  },
]
`;

exports[`custom-property-name does error with a mismatched name with hyphen 1`] = `
Array [
  Object {
    "column": 9,
    "line": 1,
    "rule": "terra/custom-property-name",
    "severity": "error",
    "text": "Expected --test-icon to be suffixed with -background-image (terra/custom-property-name)",
  },
]
`;
